lib/hooks/useData.tsx

```diff
@@ -0,0 +1,16 @@
+import useSWR from 'swr';
+import * as fetch from '../utils/fetch';
+
+type DataResponse<T> = {
+    data?: T;
+    isLoading: boolean;
+    isError?: Error;
+};
+
+function useData<T>(endpoint: string): DataResponse<T> {
+    const ep = `${process.env.NEXT_PUBLIC_API_URL}/${endpoint}`;
+    const { data: data, error } = useSWR<T, Error>(ep, fetch.get);
+    return { data, isLoading: !data && !error, isError: error };
+}
+
```

Comment:
Let's add a parameter called queryParams that get appended to the url. See this gist for an [example](https://gist.github.com/robertsiipola/ab9bb8d22cf5efb99194f3e9260c8aaa).

This way, we can later use methods like filters etc.

Reply:
added in 82846f0